Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -477,60 +612,72 @@ public class PBXProjGenerator {
             return !linkingAttributes.isEmpty ? ["ATTRIBUTES": linkingAttributes] : nil
         }
 
-        for dependency in targetDependencies {
-
-            let embed = dependency.embed ?? target.shouldEmbedDependencies
+        func processTargetDependency(_ dependency: Dependency, dependencyTarget: Target, embedFileReference: PBXFileElement) {
+            let dependencyLinkage = dependencyTarget.defaultLinkage
```

Comment:
We can just use the following once defaultLinkage is extracted to `PBXProductType` in Linkage.swift:
```swift
let dependencyLinkage = dependencyTarget.productType.defaultLinkage ?? .none
```